modules/admin/proc-migrating-databases-to-an-external-server.adoc

```diff
@@ -0,0 +1,93 @@
++
+[source,terminal]
+----
+oc port-forward -n <your-namespace> <pgsql-pod-name> <forward-to-port>:<forward-from-port>
+----
+Where:
+* The `<pgsql-pod-name>` variable denotes the name of a PostgreSQL pod with the format `backstage-psql--<_index>`.
+* The `<forward-to-port>` variable denotes the port of your choice to forward PostgreSQL data to.
+* The `<forward-from-port>` variable denotes the local PostgreSQL instance port, usually `5432`.
++
+.Example: Configuring port forwarding
+[source,terminal]
+----
+oc port-forward -n backstage backstage-psql-backstage1-0 15432:5432
+----
+
+. Make a copy of the `db_copy.sh` script and edit the following details based on your configuration:
+
+* `to_host`: The destination host name, for example, `<db-instance-name>.rds.amazonaws.com`
+* `to_port`: The destination port that is usually `5432`
+* `to_user`: The destination server username, for example, `postgres`
+* `from_host`: The source host name that is usually `127.0.0.1`
+* `from_port`: The source port number, that is, the `<forward-to-port>` variable 
+* `from_user`: The source server username, for example, `postgres`
+* `allDB`: The name of databases to import in double quotes separated by spaces, for example, `("backstage_plugin_app" "backstage_plugin_auth" "backstage_plugin_catalog" "backstage_plugin_permission" "backstage_plugin_scaffolder" "backstage_plugin_search")`
+
+. Create a destination database for copying the data:
++
+[source,terminal]
+----
+/bin/bash TO_PSW=<destination-db-password> /path/to/db_copy.sh
+----
+Where,
+* The `<destination-db-password>` variable denotes the password to connect to the destination database.
++
+[NOTE]
+====
+You can stop port forwarding when copying of the data is complete. Also, if you are dealing with large databases, go to the link:https://www.postgresql.org/docs/current/backup-dump.html#BACKUP-DUMP-LARGE[site] and see the details to use the compression tools. 
+====
+
+. Reconfigure your `Backstage` custom resource (CR). For more information, see link:{LinkAdminGuide}#proc-configuring-postgresql-instance-using-operator_admin-rhdh[Configuring an external PostgreSQL instance using the Operator]. 
+. Check that the following code is present at the end of your `Backstage` CR after reconfiguration:
++
+[source,yaml]
+----
+# ...
+spec:
+  database:
+    enableLocalDb: false 
+  application:
+  # ... 
+    extraFiles:
+      secrets:
+        - name: <crt-secret> 
+          key: postgres-crt.pem # key name as in <crt-secret> Secret
+    extraEnvs:
+      secrets:
+        - name: <cred-secret> 
+# ...        
+----
++
+[NOTE]
+====
+Reconfiguring the `Backstage` CR deletes the corresponding `StatefulSet` and `Pod` objects, but does not delete the `PersistenceVolumeClaim` object. Use the following command to delete the local `PersistenceVolumeClaim` object:
+
+[source,terminal]
+----
+oc -n backstage delete pvc <local-psql-pvc-name>
+----
+====
+
+. Apply the configuration changes.
+
```

Comment:
Can we add the verification step as well?

Reply:
@zdrapela, Do we have any verification step that we can add for user's reference?

Reply:
User can verify that his RHDH instance is running and contains the data that was migrated, and that `oc get pods -n <your-namespace>` returns only the `backstage-developer-hub-...` pod and doesn't contain the `backstage-psql-developer-hub-0` pod, for example. The same can be done in the OCP Console - Developer view - Topolgy.